src/main/java/hudson/plugins/git/UserRemoteConfig.java

```diff
@@ -185,7 +188,7 @@ public FormValidation doCheckUrl(@AncestorInPath Item item,
             return FormValidation.ok();
         }
 
-        private static StandardCredentials lookupCredentials(Item project, String credentialId, String uri) {
+        private static StandardCredentials lookupCredentials(@CheckForNull Item project, String credentialId, String uri) {
             return (credentialId == null) ? null : CredentialsMatchers.firstOrNull(
                         CredentialsProvider.lookupCredentials(StandardCredentials.class, project, ACL.SYSTEM,
```

Comment:
I'd rather be explicit in plugins about what to fall back to if the project is `null`... perhaps we should discuss this and https://github.com/jenkinsci/git-plugin/pull/437